static/js/edit_fields_modal.js

```diff
@@ -0,0 +1,67 @@
+import $ from "jquery";
+
+import render_edit_fields_modal from "../templates/settings/edit_fields_modal.hbs";
+
+import * as blueslip from "./blueslip";
+import * as overlays from "./overlays";
+
+/*
+    Look for edit_fields_modal in settings_users.js to see an
+    example of how to use this widget.
+
+    Some things to note:
+
+        1) We create DOM on the fly, and we remove the DOM once
+           it's closed.
+
+        2) We attach the DOM for the modal to modal_fields.parent.
+
+        3) The cancel button is driven by bootstrap.js.
+
+        4) We do not handle closing the modal on clicking "Save
+           changes" here because some modals show errors, if the
+           request fails, in the modal itself without closing.
+
+        5) Some steps are needed to be done after the modal body
+           is added to DOM, and this is handled by post_render.
+*/
```

Comment:
Have written this according to my understanding, but there may still be scope of improvement here.